### DISCUSSION TITLE
A Newbie needing help with sourcekit LSP stopping to work

### DISCUSSION THREAD

**Discussion by gmtborges** [wojciech-kulik/xcodebuild.nvim#193] [Q&A] [answered]
I'm receiving this message in Neovim:
```
Client sourcekit quit with exit code 0 and signal 5. Check log for errors: /Users/gustavo/.local/state/nvim/lsp.log
```

And in the lsp.log have this
```
[START][2024-10-04 20:31:39] LSP logging initiated
[ERROR][2024-10-04 20:31:39] .../vim/lsp/rpc.lua:770    "rpc"   "/usr/bin/sourcekit-lsp"        "stderr"        "INFO:server:Xcode Build Server Startup\nINFO:server:use flags from /Users/gustavo/Library/Caches/xcode-build-server/-Users-gustavo-repos-study-MyApp/compile_file-MyApp-b23b7f5452150c2cf8ec24a67cd35a64\n"
```

**Comment by wojciech-kulik** [comment]
Please check ios-dev-starter repo and see if that config works for you to eliminate config error :).

Also, it looks like you are using `/usr/bin/sourcekit-lsp`. Probably it would be safer to use the one from Xcode:
```
cmd = vim.trim(vim.fn.system("xcrun -f sourcekit-lsp")),
```

**Comment by gmtborges** [comment]
The error happens exactly when I ran the picker command `:XcodebuildPicker<cr>`

Obs: I used Tuist to initialize the project.

**Reply by gmtborges** [reply]
I deleted everything and started again, now the errors have changed. But still triggered for the same reason `:XcodebuildPicker<cr>`

```
[ERROR][2024-10-05 09:53:19] .../vim/lsp/rpc.lua:770    "rpc"   "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp"      "stderr"        "[2024-10-05 09:53:19.604] could not find manifest, or not a SwiftPM package: /Users/gustavo/repos/study/MyApp\n"
[ERROR][2024-10-05 09:53:19] .../vim/lsp/rpc.lua:770    "rpc"   "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp"      "stderr"        "[2024-10-05 09:53:19.830] could not open compilation database for /Users/gustavo/repos/study/MyApp/MyApp/Sources/ContentView.swift\n"
[ERROR][2024-10-05 09:53:53] .../vim/lsp/rpc.lua:770    "rpc"   "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp"      "stderr"        '[2024-10-05 09:53:53.394] failed to construct TextEdits from response [\n  {\n    key.offset: 257,\n    key.length: 0,\n    key.sourcetext: "<#PrimitiveButtonStyleConfiguration#>"\n  }\n]\n'
```

**Reply by wojciech-kulik** [reply]
Have you tried creating a new project using Xcode and see if it works there? 

I can't help with `sourcekit-lsp` nor `xcode-build-server` as it is not a part of this plugin and I don't have any super knowledge about them. 

I think you should create a ticket in one of those repos, maybe they will be able to help you investigate the issue. @SolaWing from xcode-build-server has a great knowledge on how things work regarding the LSP :).

**Reply by SolaWing** [reply]
this 

> I deleted everything and started again, now the errors have changed. But still triggered for the same reason `:XcodebuildPicker<cr>`
> 
> ```
> [ERROR][2024-10-05 09:53:19] .../vim/lsp/rpc.lua:770    "rpc"   "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp"      "stderr"        "[2024-10-05 09:53:19.604] could not find manifest, or not a SwiftPM package: /Users/gustavo/repos/study/MyApp\n"
> [ERROR][2024-10-05 09:53:19] .../vim/lsp/rpc.lua:770    "rpc"   "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp"      "stderr"        "[2024-10-05 09:53:19.830] could not open compilation database for /Users/gustavo/repos/study/MyApp/MyApp/Sources/ContentView.swift\n"
> [ERROR][2024-10-05 09:53:53] .../vim/lsp/rpc.lua:770    "rpc"   "/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/sourcekit-lsp"      "stderr"        '[2024-10-05 09:53:53.394] failed to construct TextEdits from response [\n  {\n    key.offset: 257,\n    key.length: 0,\n    key.sourcetext: "<#PrimitiveButtonStyleConfiguration#>"\n  }\n]\n'
> ```

This error looks like buildServer.json is not recognized and started normally.
Also, can you compile this normally in xcode?

PS: You can export environment `SOURCEKIT_LOGGING=3` to LSP, to view more detailed logs

**Reply by gmtborges** [reply]
Actually, I was following a wrong lead.
The sourcekit-lsp client crashes after any run command (Telescope, Explorer, Xcodebuild, ...).
There is a crash log in the Console app, but I couldn't troubleshoot it.

I think the problem is isolated with sourcekit-lsp. There is nothing related to xcode-build-server and xcodebuild.nvim

ChatGPT told me this:

From the crash report, it appears that sourcekit-lsp is crashing due to an EXC_BREAKPOINT (SIGTRAP) signal. This is a trap typically used for debugging purposes, indicating that sourcekit-lsp encountered an issue serious enough to trigger a forced termination. The key details from the log:

Exception Type: EXC_BREAKPOINT (SIGTRAP) — The crash occurred when sourcekit-lsp encountered a trap, likely due to an internal assertion failure or a condition it couldn't recover from.

Crash Location: The crash occurs in sourcekit-lsp at the memory address 0x00000001043ebe9c, which is deep within the sourcekit-lsp executable. It's challenging to diagnose the exact cause without further insights into the code, but it suggests that a specific condition caused it to fail.

Termination Reason: The crash is classified under the SIGNAL namespace, specifically for SIGTRAP, meaning sourcekit-lsp hit a critical issue that it could not handle.

Potential Causes:
Xcode Compatibility: There could be an issue with the version of sourcekit-lsp bundled with Xcode. If you're using an older or newer Xcode version, this might be the root cause. You could try updating or downgrading Xcode to see if that resolves the issue.

Neovim LSP Integration: The way Neovim's LSP client is interacting with sourcekit-lsp might be causing problems. If Neovim is sending malformed requests or commands, it could lead to sourcekit-lsp failing.

**Reply by Woit** [reply]
Yes, sourcekit may quite unstable at this time. Especially when you use some kind of autocompletion (nvim-cmp, or copilot). For now i "solve" it in a dumb way - just defined keymap for quick reload lsp

**Reply by gmtborges** [reply, marked as answer]
I found the problem, it's this Neovim plugin `brenoprata10/nvim-highlight-colors`, but I don't know why.
